src/lib.rs

```diff
@@ -62,13 +65,15 @@
 #![no_std]
 #![cfg_attr(feature = "internal_benches", allow(unstable_features), feature(test))]
 
-#[cfg(any(test, feature = "use_heap"))]
+#[cfg(any(test, feature = "alloc"))]
```

Comment:
`#[cfg(feature = "alloc")]` is all that's needed here, as `test` requires `alloc`.

In general, there a lot of `#[cfg(any(test, feature = "alloc"))]` in the code. These should be simplified to `#[cfg( feature = "alloc")]